File: 119_PascalTriangle_II.py
# 36.8%
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        res = [1]
        for i in range(0, rowIndex):
            res = map(lambda x, y: x + y, res + [0], [0] + res)
        return res

File: 119_PascalTriangle_II.cpp
//1.86%
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        //vector<int> curr;
		vector<int> last;
		for(auto i = 0; i <= rowIndex; i++){
			vector<int> curr(i + 1, 1);
			for(auto j = 1; j < i; j++){
				curr[j] = last[j - 1] + last[j];
			}
			last = curr;
		}
		return last;
    }
};